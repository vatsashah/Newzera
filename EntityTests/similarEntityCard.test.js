// Importing required library
import React from 'react'
import { shallow, mount } from 'enzyme'

// Importing component to be tested
import SimilarEntityCard from '../../../Components/Entity/similarEntityCard'

describe('Similar entity Card should render without any error', () => {
  it('Similar entity Card should render correctly', () => {
    const component = shallow(<SimilarEntityCard />)
    expect(component).toMatchSnapshot()
  })
  it('Entities in Similar Entity cards section should have same number of entity names and entity images', () => {
    const wrapper = mount(<SimilarEntityCard />).props()
    const numberOfEntityNames = wrapper.entityName.length
    const numberOfEntityImages = wrapper.entityImg.length
    expect(numberOfEntityImages).toEqual(numberOfEntityNames)
  })
})
