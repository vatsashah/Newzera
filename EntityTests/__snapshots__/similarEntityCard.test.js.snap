// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Similar entity Card should render without any error Similar entity Card should render correctly 1`] = `
<div
  className="SimilarEntityCardContainer"
  style={
    Object {
      "left": "0vw",
      "top": "0vh",
    }
  }
>
  <div>
    <p
      className="SimilarEntityCardContainerHeading"
    >
      Similar Entities
    </p>
  </div>
  <div
    className="PointerWhenOnHover"
  >
    <p
      className="SimilarEntityViewAll"
    >
      View all
    </p>
  </div>
  <div
    className="Left30pxSpace"
  />
  <div
    className="Right30pxSpace"
  />
  <div
    className="row_entities"
    onScroll={[Function]}
  >
    <div
      className="ExtraSpace1"
    >
      <div
        className="ExtraSpace2"
      />
    </div>
    <div
      className="wrap"
      key="0"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="India.png"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            India
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={true}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Follow"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="1"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="usaSq.jpg"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            United States of America
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={true}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Follow"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="2"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="India.png"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            India
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={false}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Following"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="3"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="usaSq.jpg"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            United States of America
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={false}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Following"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="4"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="India.png"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            India
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={false}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Following"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="5"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="Trump.png"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            Trump
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={true}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Follow"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="6"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="India.png"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            India
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={true}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Follow"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="7"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="usaSq.jpg"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            United States of America
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={true}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Follow"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="8"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="India.png"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            India
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={false}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Following"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="9"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="usaSq.jpg"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            United States of America
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={false}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Following"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="10"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="India.png"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            India
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={false}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Following"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="wrap"
      key="11"
    >
      <div
        className="entity_card"
      >
        <div>
          <img
            alt="Entity"
            className="similarEntityImage"
            src="Trump.png"
          />
        </div>
        <div>
          <p
            className="similarEntityName"
          >
            Trump
          </p>
        </div>
        <div
          className="PointerWhenOnHover similarIsfollowing"
        >
          <RectangleButton
            bordered={true}
            disabled={false}
            filled={false}
            fontSize={15}
            height={35}
            radius={38.5}
            title="Follow"
            width={112}
          />
        </div>
      </div>
    </div>
    <div
      className="ExtraSpace1"
    >
      <div
        className="ExtraSpace2"
      />
    </div>
  </div>
  <div
    className="left_arrow scroll_button"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex="0"
  >
    <i
      className="arrow left"
    />
  </div>
  <div
    className="right_arrow scroll_button"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <i
      className="arrow right"
    />
  </div>
</div>
`;
